src/block-data-providers/bitcoin-core/provider.ts

```diff
@@ -0,0 +1,278 @@
+                params: [txid, isVerbose],
+                ...this.createPartialRPCBody(),
+            };
+            return await this.request(body);
+        } catch (error) {
+            this.logger.log(
+                `Error fetching transaction with transaction id : ${txid}`,
+            );
+            throw error;
+        }
+    }
+
+    public async processBlock(height: number): Promise<Transaction[]> {
+        const parsedTransactionList: Transaction[] = [];
+        const blockHash = await this.getBlockHash(height);
+        this.logger.log(
+            `Processing block at height ${height}, hash ${blockHash}`,
+        );
+        const block = await this.getBlock(blockHash, 2);
+        for (const txn of block.tx) {
+            const parsedTransaction = await this.parseTransaction(
+                txn,
+                block.hash,
+                block.height,
+            );
+            parsedTransactionList.push(parsedTransaction);
+        }
+        return parsedTransactionList;
+    }
+
+    private async parseTransaction(
+        txn: BlockTransaction,
+        blockHash: string,
+        blockHeight: number,
+    ): Promise<Transaction> {
+        const inputs: TransactionInput[] = await Promise.all(
+            txn.vin.map(
+                async (input) => await this.parseTransactionInput(input),
+            ),
+        );
+        const outputs: TransactionOutput[] = txn.vout.map((output) =>
+            this.parseTransactionOutput(output),
+        );
+
+        return {
+            txid: txn.txid,
+            vin: inputs,
+            vout: outputs,
+            blockHeight,
+            blockHash,
+        };
+    }
+
+    private async parseTransactionInput(
+        txnInput: Input,
+    ): Promise<TransactionInput> {
+        let txid =
+            '0000000000000000000000000000000000000000000000000000000000000000';
+        let vout = 0;
+        let prevOutScript = '';
+        let witness = undefined;
+        let scriptSig = '';
+
+        if (!isCoinbaseInput(txnInput)) {
+            txid = txnInput.txid;
+            const prevTransaction = await this.getRawTransaction(
+                txnInput.txid,
+                true,
+            );
+            vout = txnInput.vout;
+            prevOutScript = prevTransaction.vout.find((out) => out.n == vout)
+                .scriptPubKey.hex;
+            witness = txnInput.txinwitness;
+            scriptSig = txnInput.scriptSig.hex;
+        }
+
+        return {
+            txid,
+            vout,
+            scriptSig,
+            witness,
+            prevOutScript,
+        };
+    }
+
+    private parseTransactionOutput(txnOutput: Output): TransactionOutput {
+        return {
+            scriptPubKey: txnOutput.scriptPubKey.hex,
+            value: txnOutput.value,
+        };
+    }
+
+    async request(body: RPCRequestBody): Promise<any> {
+        const { rpcUser, rpcPass, rpcPort, rpcHost } = this.config;
```

Comment:
instead of copying this again from `configService` why not use the service here to get it?
so we can do something like
```ts
                {
                     auth: {
                         username: this.configService.get<string>('bitcoinCore.rpcUser'),
                         password: this.configService.get<string>('bitcoinCore.rpcPass'),
                     },
                 },
```

Reply:
I believe individual preference is at war here. I  rather prefer all config data to be be initialised at the constructor, this helps me to localise any alteration that might be done which pertains to configurations.